src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1610,26 +1610,35 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
 
     if(!perform_virtual_dispatch(member))
     {
-      exprt base;
-      if(get_expr(*member.getBase(), base))
-        return true;
+      if(!is_member_decl_static(member))
+      {
+        exprt base;
+        if(get_expr(*member.getBase(), base))
+          return true;
 
-      exprt comp;
-      if(get_decl(*member.getMemberDecl(), comp))
-        return true;
+        exprt comp;
+        if(get_decl(*member.getMemberDecl(), comp))
+          return true;
 
-      if(!comp.name().empty())
-      {
-        // for MemberExpr referring to struct field (an/or method in case of C++)
+        assert(!comp.name().empty());
+        // for MemberExpr referring to struct field (or method in case of C++ class)
         new_expr = member_exprt(base, comp.name(), comp.type());
       }
       else
       {
-        // for MemberExpr in referring to a static member
-        // which is essentially a VarDecl
+        // if a MemberExpr refers to a static member decl,
+        // we directly get the member decl symbol.
+        exprt comp;
+        if(get_decl(*member.getMemberDecl(), comp))
```

Comment:
this if is duplicated in both branches. It should probably be moved to outside the branches.